primer_designer/views.py

```diff
@@ -138,26 +137,21 @@ def call_primer_designer(request, regions_form, primer_path, cmd):
         traceback = e.stderr.decode('utf-8').rstrip('\n')
         if 'Error' in traceback:
```

Comment:
this seems terrible (not sure if it was me or has always been there), probably would be better to check the return code of the call to assert if there was an error instead of parsing a traceback. Need to check with how primer designer runs if there's an error it actually returns a non zero exit code too because it has loads of subprocess fun in itself

Reply:
We did get the error code in line 156 below `e.returncode` - which is non-zero exit status 1. The traceback parsing is mainly to get the right error message to display to the front-end. Below is the error message and code passed to the frontend if primer designer subprocess crashed halfway / primer3 or smalt failed
![Screenshot from 2022-06-17 13-39-12](https://user-images.githubusercontent.com/60819172/174299914-0e9e63c3-abcf-4299-99d2-36d9be8c93a2.png)
.